### PR TITLE
[install scripts] Add support of DD_SITE

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

Adds support of `DD_SITE` env var in install scripts. Follow-up to https://github.com/DataDog/datadog-agent/pull/2412, see details there

### Additional Notes

Should be merged once 6.6.0 is out

### CODE REVIEW

File: cmd/agent/install_script.sh [hunk 1]
```diff
@@ -224,6 +228,10 @@ else
       no_start=true
     fi
   fi
+  if [ $site ]; then
```
Comment (hush-hush) [inline cmd/agent/install_script.sh]: Why user `$site` and note directly `$DD_SITE` ?

File: cmd/agent/install_mac_os.sh [hunk 2]
```diff
@@ -25,6 +25,10 @@ if [ -n "$DD_API_KEY" ]; then
     apikey=$DD_API_KEY
 fi
 
+if [ -n "$DD_SITE" ]; then
```
Comment (hush-hush) [inline cmd/agent/install_mac_os.sh]: Same here.

File: cmd/agent/install_script.sh (hunk 1 above)
Comment (olivielpeau) [reply cmd/agent/install_script.sh]: to improve a little bit the readability and maintainability of the script, all the supported env vars are listed once at the top of the script, and copied to script-local variables (and I think that's good practice)

Comment (hush-hush) [reply cmd/agent/install_script.sh]: OK :+1:

**Review by hush-hush — DISMISSED** [review]

**Review by arbll — APPROVED** [review]
🇪🇺

**Pull request merged**